x-pack/plugins/transform/server/routes/api/privileges.ts

```diff
@@ -8,65 +8,80 @@
 import type { IScopedClusterClient } from '@kbn/core/server';
 import type { SecurityHasPrivilegesResponse } from '@elastic/elasticsearch/lib/api/types';
 import { getPrivilegesAndCapabilities } from '../../../common/privilege/has_privilege_factory';
-import { APP_CLUSTER_PRIVILEGES, APP_INDEX_PRIVILEGES } from '../../../common/constants';
+import {
+  addInternalBasePath,
+  APP_CLUSTER_PRIVILEGES,
+  APP_INDEX_PRIVILEGES,
+} from '../../../common/constants';
 import type { Privileges } from '../../../common/types/privileges';
 
 import type { RouteDependencies } from '../../types';
-import { addBasePath } from '..';
 
 export function registerPrivilegesRoute({ router, license }: RouteDependencies) {
-  router.get(
-    { path: addBasePath('privileges'), validate: {} },
-    license.guardApiRoute(async (ctx, req, res) => {
-      const privilegesResult: Privileges = {
-        hasAllPrivileges: true,
-        missingPrivileges: {
-          cluster: [],
-          index: [],
-        },
-      };
+  router.versioned
+    .get({
+      path: addInternalBasePath('privileges'),
+      access: 'internal',
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {},
```

Comment:
nit: Since we are not validating here, we can change this to validate: false?

Reply:
I wasn't 100% sure if `validate: {}` is identical to `validate: false` so left this with no changes.
But it's very low risk, so I'll change it.

Reply:
Updated https://github.com/elastic/kibana/pull/158273/commits/e8cfb924d6b410d6e898f3d961422d83d56794ed